perf(minifier): exit early in `is_closest_function_scope_an_async_generator`

sapphi-red opened the pull request:
`ctx.is_in_async_generator` was searching for async generator in ancestors even though it's not needed if a normal function or an arrow function is found while searching for it.

I renamed to the method to `ctx.is_closest_function_scope_an_async_generator` and made it exit early if a normal function or an arrow function is found. This should help performance a bit in most cases as I assume there are normal functions in many places. This would also make some minor cases to be compressible (added as test cases).

Boshen commented:
### Merge activity

* **Aug 24, 1:09 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
perf(minifier): exit early in `is_closest_function_scope_an_async_generator` (#13273)

`ctx.is_in_async_generator` was searching for async generator in ancestors even though it's not needed if a normal function or an arrow function is found while searching for it.

I renamed to the method to `ctx.is_closest_function_scope_an_async_generator` and made it exit early if a normal function or an arrow function is found. This should help performance a bit in most cases as I assume there are normal functions in many places. This would also make some minor cases to be compressible (added as test cases).
